sending/src/main/java/org/kin/sending/view/RecipientAddressFragment.java

```diff
@@ -109,6 +166,33 @@ public void showAddressValidity(boolean isValid, boolean showDetail) {
     @Override
     public void updateReceiverAddress(String pasteAddress) {
         inputRecipientAddress.setText(pasteAddress);
-        //TODO scroll to the end
+    }
+
+    @Override
+    public void notifyContactChanged() {
+        listAdapter.refreshData();
+    }
+
+    @Override
+    public void updateListVisibility(final boolean isEmptyList) {
```

Comment:
You are already calling this from the main thread.
If not then change the looper to be the main thread looper.
And maybe you will even remove the background thread so maybe nothing of this is needed.
Check...